src/mpid/ch4/shm/posix/posix_init.c

```diff
@@ -148,6 +161,18 @@ static int init_vci(int vci)
     goto fn_exit;
 }
 
+static void init_topo_info(void)
+{
+    MPIDI_POSIX_topo_info_t *topo = &MPIDI_POSIX_global.topo;
+    MPIR_hwtopo_gid_t core_id = MPIR_hwtopo_get_obj_by_type(MPIR_HWTOPO_TYPE__CORE);
+    MPIR_hwtopo_gid_t l3_cache_id = MPIR_hwtopo_get_obj_by_type(MPIR_HWTOPO_TYPE__L3CACHE);
+    MPIR_hwtopo_gid_t numa_id = MPIR_hwtopo_get_obj_by_type(MPIR_HWTOPO_TYPE__PACKAGE);
```

Comment:
It looks like hwloc still defines `HWLOC_OBJ_NUMANODE`, our interface just doesn't provide an equivalent for it. Can you add support for it and use it here instead of package to avoid any possible mismatch? https://www.open-mpi.org/projects/hwloc/doc/v2.11.0/a00140.php#ggacd37bb612667dc437d66bfb175a8dc55a9d917a3e5497950c6d8948b8e183db5a

Reply:
I guess in our abstraction `MPIR_HWTOPO_TYPE__DDR` is equivalent to `HWLOC_OBJ_NUMANODE`.

Reply:
```suggestion
    MPIR_hwtopo_gid_t numa_id = MPIR_hwtopo_get_obj_by_type(MPIR_HWTOPO_TYPE__DDR);
```